src/remake.jl

```diff
@@ -668,17 +670,19 @@ function _updated_u0_p_symmap(prob, u0, ::Val{true}, p, ::Val{false}, t0)
     # This is sort of an implicit dependency on MTK. The values of `u` won't actually be
     # used, since any state symbols in the expression were substituted out earlier.
     temp_state = ProblemState(; u = state_values(prob), p = p, t = t0)
-    u0 = anydict(k => symbolic_type(v) === NotSymbolic() ? v : getu(prob, v)(temp_state)
-    for (k, v) in u0)
+    for (k, v) in u0
+        u0[k] = symbolic_type(v) === NotSymbolic() ? v : getu(prob, v)(temp_state)
+    end
     return remake_buffer(prob, state_values(prob), keys(u0), values(u0)), p
 end
 
 function _updated_u0_p_symmap(prob, u0, ::Val{false}, p, ::Val{true}, t0)
     isdep = any(symbolic_type(v) !== NotSymbolic() for (_, v) in p)
     isdep || return u0, remake_buffer(prob, parameter_values(prob), keys(p), values(p))
 
-    p = anydict(k => symbolic_type(v) === NotSymbolic() ? v : symbolic_evaluate(v, p)
-    for (k, v) in p)
+    for (k, v) in p
+        p[k] = symbolic_type(v) === NotSymbolic() ? v : symbolic_evaluate(v, p)
+    end
```

Comment:
why is this preferred? This would be a slower construction.

Reply:
Is it? I thought it would be faster since it doesn't create a new dictionary

Reply:
oh I didn't see that it was extending an old one.

It's still adding elements one by one. That might amortize?

Reply:
```julia
julia> function oop(dict)
           return Dict{Any, Any}(k => 2v for (k, v) in dict)
       end
julia> function iip(dict)
           for (k, v) in dict
               dict[k] = 2v
           end
       end
julia> dict = Dict{Any, Any}(rand() => rand() for _ in 1:10000);
julia> oop(dict); iip(dict);
julia> @be oop(dict)
Benchmark: 25 samples with 1 evaluation
 min    2.599 ms (50092 allocs: 1.183 MiB)
 median 2.696 ms (50092 allocs: 1.183 MiB)
 mean   3.925 ms (50092 allocs: 1.183 MiB, 3.60% gc time)
 max    29.143 ms (50092 allocs: 1.183 MiB, 90.08% gc time)

julia> @be iip(dict)
Benchmark: 86 samples with 1 evaluation
 min    1.004 ms (20085 allocs: 313.828 KiB)
 median 1.052 ms (20085 allocs: 313.828 KiB)
 mean   1.140 ms (20085 allocs: 313.828 KiB, 1.77% gc time)
 max    5.962 ms (20085 allocs: 313.828 KiB, 81.95% gc time)
```

In-place seems faster

Reply:
Okay, that's mainly because the generator isn't optimizing there 😅 . Both are slow, but it's fine.